### PR TITLE
Start all services in parallel inside device-steps

### PR DESCRIPTION
**Pull request by giggsoff**
Refactor pillar to be able to start all services in parallel inside device-steps

### CODE REVIEW

**Comment by giggsoff** [comment]
If I correctly understood the comment https://github.com/lf-edge/eve/pull/2839#issuecomment-1266149108, we should firstly try to refactor existence logic to be able to start services at some place in device-steps (after upgrade converter) and split device-steps later onto onboot and service part.
Inside this PR I put required changes. Probably we can split the last commit (about Fix flags parsing in go routines) into separate PR.

File: pkg/pillar/scripts/device-steps.sh [hunk 1]
```diff
@@ -379,19 +375,23 @@ $BINDIR/diag -f -o /run/diag.pipe runAsService &
 
 echo "$(date -Ins -u) Starting tpmmgr as a service agent"
 $BINDIR/tpmmgr runAsService &
-wait_for_touch tpmmgr
-touch "$WATCHDOG_FILE/tpmmgr.touch"
```
Comment (rene) [inline pkg/pillar/scripts/device-steps.sh]: tpmmgr is not in AGENTS0, shouldn't be added in order to get .touch watchdog file created later?

File: pkg/pillar/cmd/ipcmonitor/ipcmonitor.go [hunk 2]
```diff
@@ -29,24 +29,29 @@ import (
...
-	topicPtr := flag.String("t", "DeviceNetworkStatus",
+	agentScopePtr := flagSet.String("s", "", "agentScope")
+	topicPtr := flagSet.String("t", "DeviceNetworkStatus",
 		"topic")
-	debugPtr := flag.Bool("d", false, "Debug flag")
-	persistentPtr := flag.Bool("P", false, "Persistent flag")
-	formatPtr := flag.String("f", "go", "format flag, defaults to 'go', supports: 'go', 'json'")
-	flag.Parse()
-	agentName := *agentNamePtr
+	debugPtr := flagSet.Bool("d", false, "Debug flag")
```
Comment (rene) [inline pkg/pillar/cmd/ipcmonitor/ipcmonitor.go:48]: This code excerpt it's been repeated a lot throughout the files, which includes a lot of same strings ("Debug flag", "Persistent flag", etc), would be worth to create a function (or other elegant solution)  to reduce this repetition?

Comment (giggsoff) [reply pkg/pillar/cmd/ipcmonitor/ipcmonitor.go:48]: Good point. And we have one in AgentBase. Let me try

File: pkg/pillar/scripts/device-steps.sh (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/scripts/device-steps.sh]: Nice catch, thank you. Hope it did not stopped in my experiments because of this issue.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
Since we are starting all the agents before we run client we might not have a deviceUUID or device certificate when those services start.
We can do this but to be safe we should add WaitForOnboarded() to the top of Run in agents which are currently started after client has run.

**Comment by giggsoff** [comment]
> We can do this but to be safe we should add WaitForOnboarded() to the top of Run in agents which are currently started after client has run.

Well, seems it works, and handle certificate generation order as well (as we start client after all certs generated), thank you!

**Review by eriknordmark — APPROVED** [review]
Let's run this through the tests. We probably want to do some additional tests as well to make sure we don't have races in corner cases.
